web-common/src/features/dashboards/time-series/MetricsTimeSeriesCharts.svelte

```diff
@@ -247,6 +249,34 @@
   const setAllMeasuresVisible = () => {
     showHideMeasures.setAllToVisible();
   };
+
+  function onKeyDown(e) {
+    if (scrubStart && scrubEnd) {
+      // if key Z is pressed, zoom the scrub
+      if (e.key === "z") {
+        const { start, end } = getOrderedStartEnd(
+          $dashboardStore?.selectedScrubRange?.start,
+          $dashboardStore?.selectedScrubRange?.end
+        );
+        metricsExplorerStore.setSelectedTimeRange(metricViewName, {
```

Comment:
Why not merge these actions instead of using setTimeout?

Reply:
I tried merging it into a single dashboard reducer call but the two values (`timeRange` and `scrubRange`) are serialized independently on the protos. Successive calls without any delay leads to race condition.